llama-rs/src/llama.rs

```diff
@@ -105,32 +124,108 @@ fn llama_n_parts(size: i32) -> i32 {
     }
 }
 
+/// Each variant represents a step within the process of loading the model.
+/// These can be used to report progress to the user.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub enum LoadProgress<'a> {
```

Comment:
A very clean solution! :smile:  Having this will be especially important if we end up implementing parallel loading as described on #15. Because relying on the order of print messages for the little dots breaks the moment you start multithreading.